File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/UserService.java
package org.mifos.mobilewallet.core.data.fineract.api.services;

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints;
import org.mifos.mobilewallet.core.data.fineract.api.GenericResponse;
import org.mifos.mobilewallet.core.data.fineract.entity.UserWithRole;
import org.mifos.mobilewallet.core.domain.model.user.NewUser;
import org.mifos.mobilewallet.core.domain.usecase.user.CreateUser;

import java.util.List;

import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;
import rx.Observable;

/**
 * Created by ankur on 11/June/2018
 */

public interface UserService {

    @GET(ApiEndPoints.USER)
    Observable<List<UserWithRole>> getUsers();

    @POST(ApiEndPoints.USER)
    Observable<CreateUser.ResponseValue> createUser(@Body NewUser user);

    @PUT(ApiEndPoints.USER + "/{userId}")
    Observable<GenericResponse> updateUser(
            @Path("userId") int userId,
            @Body Object updateUserEntity);

    @DELETE(ApiEndPoints.USER + "/{userId}")
    Observable<GenericResponse> deleteUser(
            @Path("userId") int userId);

    @GET(ApiEndPoints.USER + "/{userId}")
    Observable<UserWithRole> getUser(
            @Path("userId") long userId);
}

File: core/src/main/java/org/mifos/mobilewallet/core/data/fineract/api/services/UserService.kt
package org.mifos.mobilewallet.core.data.fineract.api.services

import org.mifos.mobilewallet.core.data.fineract.api.ApiEndPoints
import org.mifos.mobilewallet.core.data.fineract.entity.UserWithRole
import org.mifos.mobilewallet.core.domain.model.user.NewUser
import org.mifos.mobilewallet.core.domain.usecase.user.CreateUser
import org.mifos.mobilewallet.core.data.fineract.api.GenericResponse
import retrofit2.http.*
import rx.Observable

/**
 * Created by ankur on 11/June/2018
 */
interface UserService {
    @get:GET(ApiEndPoints.USER)
    val users: Observable<List<UserWithRole?>?>?

    @POST(ApiEndPoints.USER)
    fun createUser(@Body user: NewUser?): Observable<CreateUser.ResponseValue?>?

    @PUT(ApiEndPoints.USER + "/{userId}")
    fun updateUser(
        @Path("userId") userId: Int,
        @Body updateUserEntity: Any?
    ): Observable<GenericResponse?>?

    @DELETE(ApiEndPoints.USER + "/{userId}")
    fun deleteUser(
        @Path("userId") userId: Int
    ): Observable<GenericResponse?>?

    @GET(ApiEndPoints.USER + "/{userId}")
    fun getUser(
        @Path("userId") userId: Long
    ): Observable<UserWithRole?>?
}